mujoco_warp/_src/collision_driver.py

```diff
@@ -35,6 +50,56 @@
 wp.set_module_options({"enable_backward": False})
 
 
+_CONVEX_COLLISION_PAIRS = {
+  (GeomType.PLANE, GeomType.SPHERE): plane_sphere_wrapper,
+  (GeomType.PLANE, GeomType.CAPSULE): plane_capsule_wrapper,
+  (GeomType.PLANE, GeomType.ELLIPSOID): plane_ellipsoid_wrapper,
+  (GeomType.PLANE, GeomType.CYLINDER): plane_cylinder_wrapper,
+  (GeomType.PLANE, GeomType.BOX): plane_box_wrapper,
+  (GeomType.PLANE, GeomType.MESH): plane_convex_wrapper,
+  (GeomType.HFIELD, GeomType.SPHERE): None,
+  (GeomType.HFIELD, GeomType.CAPSULE): None,
+  (GeomType.HFIELD, GeomType.ELLIPSOID): None,
+  (GeomType.HFIELD, GeomType.CYLINDER): None,
+  (GeomType.HFIELD, GeomType.BOX): None,
+  (GeomType.HFIELD, GeomType.MESH): None,
+  (GeomType.SPHERE, GeomType.SPHERE): sphere_sphere_wrapper,
+  (GeomType.SPHERE, GeomType.CAPSULE): sphere_capsule_wrapper,
+  (GeomType.SPHERE, GeomType.ELLIPSOID): None,
+  (GeomType.SPHERE, GeomType.CYLINDER): sphere_cylinder_wrapper,
+  (GeomType.SPHERE, GeomType.BOX): sphere_box_wrapper,
+  (GeomType.SPHERE, GeomType.MESH): None,
+  (GeomType.CAPSULE, GeomType.CAPSULE): capsule_capsule_wrapper,
+  (GeomType.CAPSULE, GeomType.ELLIPSOID): None,
+  (GeomType.CAPSULE, GeomType.CYLINDER): None,
+  (GeomType.CAPSULE, GeomType.BOX): capsule_box_wrapper,
+  (GeomType.CAPSULE, GeomType.MESH): None,
+  (GeomType.ELLIPSOID, GeomType.ELLIPSOID): None,
+  (GeomType.ELLIPSOID, GeomType.CYLINDER): None,
+  (GeomType.ELLIPSOID, GeomType.BOX): None,
+  (GeomType.ELLIPSOID, GeomType.MESH): None,
+  (GeomType.CYLINDER, GeomType.CYLINDER): None,
+  (GeomType.CYLINDER, GeomType.BOX): None,
+  (GeomType.CYLINDER, GeomType.MESH): None,
+  (GeomType.BOX, GeomType.BOX): box_box_wrapper,
+  (GeomType.BOX, GeomType.MESH): None,
+  (GeomType.MESH, GeomType.MESH): None,
+}
+
+
+def _check_convex_collision_pairs():
+  prev_idx = -1
+  for pair in _CONVEX_COLLISION_PAIRS.keys():
+    idx = upper_trid_index(len(GeomType), pair[0].value, pair[1].value)
+    if pair[1] < pair[0] or idx <= prev_idx:
+      return False
+    prev_idx = idx
+  return True
+
+
+assert _check_convex_collision_pairs(), "_CONVEX_COLLISION_PAIRS is in invalid order."
```

Comment:
This is cool, I like it a lot.  How about we make it a unit test instead?  The main advantage is just to save users from running this on module import.  I have similar unit tests that test static properties of the code, so we have precedent - check types_test.py for example.